### ISSUE TITLE
Include paths from c_cpp_properties.json are not passed to clang-tidy for code analysis

### ISSUE THREAD

**Issue by psclkhoury** [microsoft/vscode-cpptools#9032]
I have a `c_cpp_properties.json` file defined in the repo and it contains quite a few entries under `includePath`.

Those include paths are not passed to clang-tidy when I run the code analysis and so I always have errors with not found header files.

I see that the `defines` from `c_cpp_properties.json` are passed in but not the includes.

It would be nice if all the configuration in `c_cpp_properties.json` is used to make clang-tidy work without having to redefine them somewhere else.

**Comment by sean-mcmanus** [comment]
The current implementation is supposed to pass all the include paths that are used by IntelliSense, so what you're seeing is not expected unless you have something like configurationProvider or compileCommands set that would cause IntelliSense to not use the includePath property. . When you run the C/C++: Log Diagnostics command with that file active do you see the include path in the Includes list?

**Comment by psclkhoury** [comment]
I tested this again with the newest insider version (Version 1.10.0: April 21, 2022) and it still doesn't work.

I see this switch being passed to clang-tidy:

```sh
--header-filter=(C:/Path/To/MyWorkspaceFolder|C:\\Path\\To\\MyWorkspaceFolder)
```

And all the rest are `-D` switches that are taken from the `defines` entry of my `c_cpp_properties.json` file.

**Comment by sean-mcmanus** [comment]
I'm not able to repro this. I don't know why the include paths wouldn't be used. After the `-D` in the logging it should normally have the `-I` listed (unless possibly if response files are being used, i.e. https://github.com/microsoft/vscode-cpptools/issues/9102). You're not using a compile_commands.json, right?

**Comment by psclkhoury** [comment]
No I don't use a `compile_commands.json`. The only 'special' thing in my `c_cpp_properties.json` is that it has ~2300 entries under `"includePath"`. Could that be causing issues?

**Comment by sean-mcmanus** [comment]
Yes, having 2300 includePath entries on Windows will require that the command line be turned into a response file, in which case you should see `@<file>.rsp` in the logging instead of -I entries. We create the .rsp file under a temp directory. It's possible that there may be a failure to create the temp file that we have not seen before. You also may be hitting bug https://github.com/microsoft/vscode-cpptools/issues/9102, in which case a workaround would be to set "C_Cpp.codeAnalysis.maxConcurrentThreads": 1...if that doesn't work then you're hitting a different issue.

**Comment by psclkhoury** [comment]
Okay now I see that there is indeed a `@<file>.rsp` being passed to clang-tidy. I added `"C_Cpp.codeAnalysis.maxConcurrentThreads": 1` but I still got the same problem. The `.rsp` file is deleted quite quickly so I don't have time to check its content.

**Comment by sean-mcmanus** [comment]
Yeah, on 2nd look it looks like we fail to handle "/I" (only "-I") and we fail to handle spaces in the paths (missing quotes), so I'm guessing you're hitting one of those cases.

**Comment by sean-mcmanus** [comment]
Do you repro this bug with 1.9.8? The bug quotes appears like it would only repro with 1.10.0, so if 1.9.8 is failing then it seems like there could be another bug as well.

**Comment by sean-mcmanus** [comment]
Well, the original bug was logged before 1.10.0 was released, so it looks like you did repro it with 1.9.7/1.9.8. It's possible you're hitting an issue that was fixed in 1.10.0. I'll also add Debug level logging of the response file to make further diagnostics easier.

**Comment by sean-mcmanus** [comment]
Yikes, I'm seeing the response files not working at all currently -- it turns out our tests were passing because the includes were getting pulled from the environment instead of the response files :(

**Comment by sean-mcmanus** [comment]
Looks like our extension never worked with response files, but I got it to work via using --extra-arg in the response file. I'm surprised it took us so long to figure this out, i.e. not enough users complained about it, and our testing was insufficient to notice the issue.

**Comment by sean-mcmanus** [comment]
This is fixed with 1.10.2 (pre-release).

**Comment by psclkhoury** [comment]
Tested it and it works! thanks.
